### PR TITLE
Support local datetime

### LINKED ISSUE #6166: support LocalDateTime feature
Since current impl is not support java8 LocalDateTime, should we add the feature to support LocalDateTime?  I can try to fix this

### PR DESCRIPTION
**Pull request by xbkaishui**
Fixes #6166.

Changes proposed in this pull request:
-
-
-
support local datetime

### CODE REVIEW

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ShardingSphereResultSet.java [hunk 1]
```diff
@@ -372,4 +376,21 @@ public Object getObject(final String columnLabel) throws SQLException {
         int columnIndex = columnLabelAndIndexMap.get(columnLabel);
         return mergeResultSet.getValue(columnIndex, Object.class);
     }
+
+    @Override
+    public <T> T getObject(final int columnIndex, final Class<T> type) throws SQLException {
```
Comment (tuohai666) [inline shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ShardingSphereResultSet.java:381]: Is it possible to support more types?

File: .gitignore [hunk 2]
```diff
@@ -35,3 +35,4 @@ logs/
 # system ignore
 .DS_Store
 Thumbs.db
+*.factorypath
```
[the remark this answers is not in the document]
Comment (xbkaishui) [reply .gitignore:38]: this is auto generate by import, not related to this project. should I unignore it ?

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ShardingSphereResultSet.java (hunk 1 above)
Comment (xbkaishui) [reply shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ShardingSphereResultSet.java:381]: good point. I will add more type support in another  pull request

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (tuohai666) [reply .gitignore:38]: I prefer remove it if this kind of files don't exist in project.

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ShardingSphereResultSet.java (hunk 1 above)
Comment (xbkaishui) [reply shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ShardingSphereResultSet.java:381]: ok, will remove it

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (xbkaishui) [reply .gitignore]: ok, will remove it

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ResultSetUtil.java [hunk 3]
```diff
@@ -20,6 +20,10 @@
 import com.google.common.primitives.Ints;
 import com.google.common.primitives.Longs;
 import com.google.common.primitives.Shorts;
+import java.time.LocalDate;
```
Comment (tuohai666) [inline shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ResultSetUtil.java:23]: Add test case in ResultSetUtilTest

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ShardingSphereResultSet.java [hunk 4]
```diff
@@ -17,6 +17,10 @@
 
 package org.apache.shardingsphere.driver.jdbc.core.resultset;
 
+import java.sql.SQLFeatureNotSupportedException;
```
Comment (tuohai666) [inline shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ShardingSphereResultSet.java:20]: Add test case in ShardingSphereResultSetTest

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ResultSetUtil.java (hunk 3 above)
Comment (xbkaishui) [reply shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ResultSetUtil.java:23]: done

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ShardingSphereResultSet.java (hunk 4 above)
Comment (xbkaishui) [reply shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ShardingSphereResultSet.java:20]: done

**Comment by EcoRIIHC** [comment]
latest version in maven repo still not support ?

**Comment by xbkaishui** [comment]
you should use the master branch.

**Comment by EcoRIIHC** [comment]
> you should use the master branch.

why not update maven repo?

**Comment by xbkaishui** [comment]
Has not been released ..

**Comment by EcoRIIHC** [comment]
> Has not been released ..

fine, got a plan ?

**Comment by xbkaishui** [comment]
@tristaZero , Do we have a plan to release 5.0 ?

**Comment by tristaZero** [comment]
I suppose the release time will be in next week. :-)

**Comment by alterhu2020** [comment]
Does this release now ?

**Comment by tristaZero** [comment]
@alterhu2020 Sure.

**Comment by JasonMing** [comment]
I cannot find any releases on maven central repo.
The latest stable version is still 4.1.1.
Can anyone release a patch version to fix this issue?

**Comment by tristaZero** [comment]
Hi 5.0.0-alpha is waiting for you, isn't it?

**Pull request merged**